dbms/src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -427,7 +427,8 @@ DictionaryConfigurationPtr getDictionaryConfigurationFromAST(const ASTCreateQuer
 
     AutoPtr<Poco::XML::Element> name_element(xml_document->createElement("name"));
     current_dictionary->appendChild(name_element);
-    AutoPtr<Text> name(xml_document->createTextNode(query.database + "." + query.table));
+    String full_name = (!database_name.empty() ? database_name : query.database) + "." + query.table;
```

Comment:
Seems like it's already substituted in InterpreterCreateQuery https://github.com/clickhouse/ClickHouse/blob/master/dbms/src/Interpreters/InterpreterCreateQuery.cpp#L714.

Reply:
It's substituted, but the result of that substitution (i.e. database's name) isn't passed to `createDictionary()`. Seems it's better to set `create.database` [there](https://github.com/clickhouse/ClickHouse/blob/master/dbms/src/Interpreters/InterpreterCreateQuery.cpp#L714) than this solution, I'll simplify the code in a separate PR.